CMakeLists.txt

```diff
@@ -67,6 +68,10 @@ if(USE_SBRK)
   add_definitions(-DUSE_SBRK)
 endif()
 
+if(EXPOSE_EXTERNAL_ALLOCATOR_API)
+  add_definitions(-DSNMALLOC_EXPOSE_RESERVE -DSNMALLOC_EXPOSE_PAGEMAP)
```

Comment:
Do we want expose reserve and expose external pagemap to be in the same define?  It might not matter for the shim library - I'm currently only exposing pagemap, when I want to reuse the Pal, I'm including the relevant `pal_{foo}.h` files and using the one for the host platform.